webapp/src/main/java/com/box/l10n/mojito/service/tm/IntegrityCheckFailureService.java

```diff
@@ -0,0 +1,140 @@
+package com.box.l10n.mojito.service.tm;
+
+import static java.util.Optional.ofNullable;
+
+import com.box.l10n.mojito.entity.TMTUVIntegrityCheckFailure;
+import com.box.l10n.mojito.entity.TMTextUnitCurrentVariant;
+import com.box.l10n.mojito.entity.TMTextUnitVariant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class IntegrityCheckFailureService {
+  private final TMTextUnitVariantRepository textUnitVariantRepository;
+
+  private final TMTUVIntegrityCheckFailureRepository integrityCheckFailureRepository;
+
+  private final TMTextUnitCurrentVariantRepository textUnitCurrentVariantRepository;
+
+  private final IntegrityCheckFailureAlertService alertService;
+
+  public IntegrityCheckFailureService(
+      TMTextUnitVariantRepository textUnitVariantRepository,
+      TMTUVIntegrityCheckFailureRepository integrityCheckFailureRepository,
+      TMTextUnitCurrentVariantRepository textUnitCurrentVariantRepository,
+      IntegrityCheckFailureAlertService alertService) {
+    this.textUnitVariantRepository = textUnitVariantRepository;
+    this.integrityCheckFailureRepository = integrityCheckFailureRepository;
+    this.textUnitCurrentVariantRepository = textUnitCurrentVariantRepository;
+    this.alertService = alertService;
+  }
+
+  @Transactional
```

Comment:
`@Transactional` on a private method won't create a translation as Spring applies this via proxies and doesn't wrap private/internal calls, if this is called from a public method marked with transactional then it will execute within the same transaction context but the annotation here has no effect